app/src/main/java/org/sopt/sample/data/remote/NetworkState.kt

```diff
@@ -0,0 +1,7 @@
+package org.sopt.sample.data.remote
+
+sealed class NetworkState {
+    object Success : NetworkState()
+    object Failure : NetworkState()
+    class Error(val e: Throwable) : NetworkState()
+}
```

Comment:
```suggestion
sealed class NetworkState {
    object Success : NetworkState()
    object Failure : NetworkState()
    data class Error(val e: Throwable) : NetworkState()
}
```
해당 error 클래스는 Throwable을 담기 위해 만든 class 이고, 
data class는 `toString`, `equals` , `copy`, `hashcode`를 자동으로 만들어주기 때문에
class 보다 data class로 만드는 것이 더 좋을 것 같습니다.